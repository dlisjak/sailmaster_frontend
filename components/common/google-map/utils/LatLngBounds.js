import LatLng from "./LatLng";

export default class LatLngBounds {
  static convert(a) {
    if (!a || a instanceof LatLngBounds) return a;
    return new LatLngBounds(a);
  }

  constructor(sw, ne) {
    if (!sw) return;

    const latlngs = ne ? [sw, ne] : sw;

    for (let i = 0, len = latlngs.length; i < len; i++) {
      this.extend(latlngs[i]);
    }
  }

  extend(obj) {
    const sw = this._sw;
    const ne = this._ne;
    let sw2;
    let ne2;

    if (obj instanceof LatLng) {
      sw2 = obj;
      ne2 = obj;
    } else if (obj instanceof LatLngBounds) {
      sw2 = obj._sw;
      ne2 = obj._ne;

      if (!sw2 || !ne2) return this;
    } else {
      return obj
        ? this.extend(LatLng.convert(obj) || LatLngBounds.convert(obj))
        : this;
    }

    if (!sw && !ne) {
      this._sw = new LatLng(sw2.lat, sw2.lng);
      this._ne = new LatLng(ne2.lat, ne2.lng);
    } else {
      sw.lat = Math.min(sw2.lat, sw.lat);
      sw.lng = Math.min(sw2.lng, sw.lng);
      ne.lat = Math.max(ne2.lat, ne.lat);
      ne.lng = Math.max(ne2.lng, ne.lng);
    }

    return this;
  }

  getCenter() {
    return new LatLng(
      (this._sw.lat + this._ne.lat) / 2,
      (this._sw.lng + this._ne.lng) / 2
    );
  }

  getSouthWest() {
    return this._sw;
  }

  getNorthEast() {
    return this._ne;
  }

  getNorthWest() {
    return new LatLng(this.getNorth(), this.getWest());
  }

  getSouthEast() {
    return new LatLng(this.getSouth(), this.getEast());
  }

  getWest() {
    return this._sw.lng;
  }

  getSouth() {
    return this._sw.lat;
  }

  getEast() {
    return this._ne.lng;
  }

  getNorth() {
    return this._ne.lat;
  }
}
