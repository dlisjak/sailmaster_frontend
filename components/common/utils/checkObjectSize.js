export default function checkObjectSize(data) {
  let formBody = [];

  for (let property in data) {
    const encodedKey = property;
    const encodedValue = data[property];
    formBody.push(encodedKey + "=" + encodedValue);
  }

  return formBody.length > 0 ? true : false;
}
