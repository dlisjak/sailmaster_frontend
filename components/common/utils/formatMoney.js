export default function formatMoney(n, c, d, t) {
  if (typeof n === "string") {
    n = parseFloat(n);
  }

  c = isNaN((c = Math.abs(c))) ? 2 : c;
  d = d === undefined ? "," : d;
  t = t === undefined ? "." : t;
  let s = n < 0 ? "-" : "";
  let i = String(parseInt((n = Math.abs(Number(n) || 0).toFixed(c))));
  let j = i.length > 3 ? i.length % 3 : 0;

  return (
    s +
    (j ? i.substr(0, j) + t : "") +
    i.substr(j).replace(/(\d{3})(?=\d)/g, "$1" + t) +
    (c
      ? d +
        Math.abs(n - i)
          .toFixed(c)
          .slice(2)
      : "")
  );
}
