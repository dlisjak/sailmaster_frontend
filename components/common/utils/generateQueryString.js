import { browserHistory } from "react-router-dom";

export default function generateQueryString(data) {
  let formBody = [];

  for (let property in data) {
    const encodedKey = property;
    const encodedValue = data[property];
    formBody.push(encodedKey + "=" + encodedValue);
  }

  browserHistory.push(
    browserHistory.getCurrentLocation().pathname + "?" + formBody.join("&")
  );
}
