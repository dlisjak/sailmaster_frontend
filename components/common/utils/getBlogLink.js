export default function getBlogLink(slug) {
  return `/blog/${slug}`;
}
