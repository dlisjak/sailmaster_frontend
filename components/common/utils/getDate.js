import moment from "moment";

export function getDate(startDate) {
  if (startDate) {
    if (moment().weekday() === 5) {
      return moment().add(7, "days");
    } else if (moment().weekday() === 6) {
      return moment().add(6, "days");
    } else {
      return moment().startOf("week").add(5, "days");
    }
  }

  if (moment().weekday() === 5) {
    return moment().add(7, "days").add(7, "days");
  } else if (moment().weekday() === 6) {
    return moment().add(6, "days").add(7, "days");
  } else {
    return moment().startOf("week").add(5, "days").add(7, "days");
  }
}
