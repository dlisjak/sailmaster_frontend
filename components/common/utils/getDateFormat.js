import moment from "moment";

export default function getDateFormat(date) {
  if (typeof date !== "string") {
    return date.format("YYYY-MM-DD");
  }

  return moment(date).format("YYYY-MM-DD");
}
