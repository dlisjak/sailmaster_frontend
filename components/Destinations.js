import React from "react";
import Link from "next/link";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import { valuesToSearch, parseDestinations } from "utils/search_utils";
import { OFFERS_URL } from "../constants";

const Destination = ({ destination }) => {
  const style = {
    backgroundImage: `url(${destination.image.thumbnail})`,
  };
  const link =
    OFFERS_URL +
    "?" +
    valuesToSearch({
      destinations: parseDestinations(destination.destination_hash.split("|")),
    });
  return (
    <div className="destination2 destinations2__link btn btn-lg btn-primary" style={style}>
      <Link href={link}>
        {destination.name}
      </Link>
    </div>
  );
};

const Destinations = ({ items }) => {
  return (
    <div className="destinations2">
      <div className="row">
        {items.map((destination) => (
          <Col
            key={destination.id}
            sm="6"
            md="4"
            lg="3"
            className="col--destinations2"
          >
            <Destination destination={destination} />
          </Col>
        ))}
      </div>
    </div>
  );
};

export default Destinations;
