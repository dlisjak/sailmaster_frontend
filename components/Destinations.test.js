import React from "react";
import renderer from "react-test-renderer";

import Destinations from "./Destinations";
import { destinations2 } from "tests/fixtures";
import { wrapRouter } from "utils/testUtils";

test("renders correctly", () => {
  const tree = renderer
    .create(wrapRouter(<Destinations items={destinations2} />))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
