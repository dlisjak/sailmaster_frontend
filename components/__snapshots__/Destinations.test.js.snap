// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="destinations2"
>
  <div
    className="row"
  >
    <div
      className="col--destinations2 col-lg-3 col-md-4 col-sm-6"
    >
      <div
        className="destination2"
        style={
          Object {
            "backgroundImage": "url(http://localhost:8000/media/__sized__/countries/budva-Montenegro-323863-1920px-16x7-thumbnail-600x600-70.jpg)",
          }
        }
      >
        <a
          className="destinations2__link btn btn-lg btn-primary"
          href="/najem-plovil?destinations=country-149-%C4%8Crna%20Gora"
          onClick={[Function]}
        >
          ÄŒrna Gora
        </a>
      </div>
    </div>
    <div
      className="col--destinations2 col-lg-3 col-md-4 col-sm-6"
    >
      <div
        className="destination2"
        style={
          Object {
            "backgroundImage": "url(http://localhost:8000/media/__sized__/countries/Piran_-Slovenia-324247-1920px-16x7-thumbnail-600x600-70.jpg)",
          }
        }
      >
        <a
          className="destinations2__link btn btn-lg btn-primary"
          href="/najem-plovil?destinations=country-201-Slovenija"
          onClick={[Function]}
        >
          Slovenija
        </a>
      </div>
    </div>
  </div>
</div>
`;
